### ISSUE TITLE
837i requires CLM segment in Subscriber HL

### ISSUE THREAD

**Issue by hackley** [irobayna/stupidedi#144]
I'm working on a script that exports health insurance claims to EDI (both professional and institutional). The script works perfectly for professional, but when I try to export an institutional claim where the patient (`HL`) is not the same as the subscriber I get the following exception:

```
Stupidedi::Exceptions::ParseError: required loop 2300 CLAIM INFORMATION is missing from loop 2000B SUBSCRIBER HIERARCHICAL LEVEL
```

For professional claims, Stupidedi allows you to omit the `CLM` segment in the subscriber loop so that you can instead specify it under the next `HL` (patient/dependent loop). The EDI spec makes it look like the same is true for institutional, as well. I've tried a bunch of small variations with no luck – is there something obvious that I'm missing?

**Comment by irobayna** [comment]
@hackley perhaps set `strict` to false, so the transaction would generate and then use `edi-pp` and `edi-ed` to try to see the problem.

**Comment by kputnam** [comment]
Hi @hackley, I haven't looked at the specs for years, but if you have the X12 documentation and are certain that the standard for 837i doesn't require a 2300 loop, then you can change the requirement of the first segment in that loop in the `Guide` for 837i.

However, if this is a deviation from the standard made by a trading partner, I would create a new `Guide` and use that when you're parsing or generating 837i's for that trading partner. That way, you won't accidentally forget to include the 2300 in the other cases.

If the `Guide` included with stupidedi is indeed wrong, please submit a PR that fixes the 2300 requirement!

**Comment by kputnam** [comment]
@hackley, I believe this will be fixed when 831c7e370b3244a5a13bd75cec018487b99b4625 is merged. The problem was that the 2300 loop was marked `Required`, however that meant it was required under both `HL` subscriber and `HL` patient. It should've been `Situational`, like the 837P definition.

**Comment by kputnam** [comment]
Closing this issue as the fix has been released.
